src/ncurses_def.cpp

```diff
@@ -18,6 +18,7 @@
 #include "catacharset.h"
 #include "color.h"
 
+#include "game.h"
```

Comment:
The fact that so much pure output code now depends on "game.h" and `class game` worries me. This is not a good design.

Reply:
I agree. I only pulled this header to access game::init_ui. Perhaps there's a better location for it?

Reply:
We could also extend our available toolkit with signals. A signal would be an excellent way to handle this situation.

Reply:
How so? This is a dependency issue, the primary uses of signals are IPC and asynchronicity, we arent dealing with either of those here.

Reply:
When it comes down to it, we are dealing with an asynchronous event. Window resizing is an external event generated by the platform window manager, and a signal and slot approach approach let's us handle it with ease. It also let's us work around the tight coupling issue and avoid adding in another header just to run a single function.

Reply:
Just because there is asynchronous behavior happening *somewhere* doesn't mean it's suitable to use an asynchronous solution for this particular problem.
In this particular case you are avoiding moving logic out of the game module (which we are in dire need of) and creating a new header by adding a new header and replacing a simple method call with indirection through the OS.
If you want to add support for handling external signals, cool, do that.
Even if we have signal support it is not ok to do communication between different parts of the program with signals.

Reply:
No, the problem is just the new inclusion of game.h, you can just add a function that accepts no arguments and unconditionally invokes the g->init_ui(), and a later cleanup can move more code into that module.
I probably would have done it for you if I had time at the moment.